<launch>
    <node pkg="nodelet" type="nodelet" name="ICP_Node" args="standalone nodelet_ns/NICP" output="screen">

        <remap from="/input_gt"         to="/gt"/>
        <remap from="/input_imu"        to="/imu"/>
        <remap from="/input_tof_pc"     to="/points"/>
        <remap from="/input_tof_nir"    to="/image_nir"/>
        <remap from="/input_tof_depth"  to="/image_depth"/>
        <remap from="/NICP/out"         to="remapped_output"/>
        <!-- use other pipeline
        0:proposed method
        1:ICP+RANSAC
        2:Others
        -->
        <param name="icp/use_other_icp"             type="int" value="0"/>
        <param name="icp/use_orig_pts"              type="bool" value="false"/>
        <param name="icp/use_ransomdownsample_pts"  type="bool" value="true"/>
        <param name="icp/use_salient_pts"           type="bool" value="false"/>
        <param name="icp/use_robust_w"              type="bool" value="true"/>

        <!-- run in vo mode with out imu -->
        <param name="icp/vo_mode"               type="bool" value="false"/>
        <!-- Init Pose with Motion Capture system data -->
        <param name="icp/init_by_MC"           type="bool" value="true"/>
        <!-- Init Pose init at the center (0,0,0) with the IMU orientation-->
        <param name="icp/init_by_IMU"          type="bool" value="false"/>
        <!-- kf_criteria 0:use_1st_frame 1:frame count based 2:distance based-->
        <param name="icp/kf_criteria"          type="int" value="2"/>

        <param name="cam_cal_file" type="string" value="$(find tof_vio)/config/flexx.yml"/>
        <param name="icp/sailentpts_sample_count"     type="int"  value="15000"/>
        <param name="icp/target_sailentpts_num"       type="int"  value="500"/>
        <param name="icp/use_depth_grad"              type="int" value="1"/>
        <param name="icp/use_intensity_grad"          type="int" value="1"/>
        <param name="icp/use_edge_detector"           type="int" value="0"/>
        <param name="icp/use_canny"                   type="int" value="0"/>
        <param name="icp/sh_backfloor"                type="double" value="0.01"/>
        <param name="icp/sh_depth"                    type="double" value="0.07"/>
        <param name="icp/sh_grey"                     type="double" value="100"/>
        <param name="icp/sh_edge"                     type="double" value="0.015"/>

    </node>

<!--################################################################################################-->
    <node pkg="tof_vio" type="eskf" name="eskf" output="screen">
        <remap from="~imu"        to="/imu"/>
        <remap from="~vo"         to="/gt/odom_gt"/>
        <param name="eskf/ng"          type="double"  value="10"/>
        <param name="eskf/na"          type="double"  value="10"/>
        <param name="eskf/nbg"         type="double" value="10"/>
        <param name="eskf/nba"         type="double" value="10"/>
        <param name="eskf/vo_p"        type="double" value="0.1"/>
        <param name="eskf/vo_q"        type="double" value="0.1"/>
        <param name="eskf/vo_delay_ms"    type="double" value="0"/>
   </node>
<!--################################################################################################-->
<!--    <node pkg="tof_vio" type="ekf" name="ekf" output="log">
        <remap from="~imu"        to="/imu"/>
        <remap from="~vo"         to="/gt/odom_gt"/>

        <param name="ekf/ng"      type="double"  value="50"/>
        <param name="ekf/na"      type="double"  value="50"/>
        <param name="ekf/nbg"     type="double" value="20"/>
        <param name="ekf/nba"     type="double" value="20"/>
        <param name="ekf/vo_p"    type="double" value="0.02"/>
        <param name="ekf/vo_q"    type="double" value="0.02"/>
    </node>-->
<!--################################################################################################-->
</launch>

